LogKitten.java

```diff
@@ -26,44 +31,99 @@
-			System.out.println("Could not create log directory");
+			System.out.println("Could not create global log directory");
 			se.printStackTrace();
 		}
-		String filePath = LogKitten.LOG_PATH + LogKitten.timestamp() + ".log"; // Set this sessions log to /home/lvuser/logs/[current time].log
-		File file = new File(filePath);
+		for(Map.Entry<Kitten, String> i : kittenList.entrySet()) {
+			try {
+				if (!i.getKey().pathDirectory.isDirectory()) {
+					i.getKey().pathDirectory.mkdirs();
+				}
+			}
+			catch (SecurityException se) {
+				System.out.println("Could not create " + i.getKey().category + " log directory");
+				se.printStackTrace();
+			}
+		}
+		String globalPath = LogKitten.GLOBAL_PATH + LogKitten.timestamp() + ".log"; // Set this sessions log to /home/lvuser/logs/global/[current time].log
+		File global = new File(globalPath);
 		try {
 			// Create new file if it doesn't exist (this should happen)
-			file.createNewFile(); // creates if does not exist
+			global.createNewFile(); // creates if does not exist
 			// Create FileOutputStream to actually write to the file.
-			LogKitten.fileOutput = new BufferedOutputStream(new FileOutputStream(file));
+			LogKitten.globalOutput = new BufferedOutputStream(new FileOutputStream(global));
 		}
 		catch (IOException ioe) {
-			System.out.println("Could not open logfile");
+			System.out.println("Could not open global logfile");
 			ioe.printStackTrace();
 		}
-		File logAlias = new File(LogKitten.LOG_ALIAS_PATH);
+		for(Map.Entry<Kitten, String> i : kittenList.entrySet()) {
+			try {
+				i.getKey().logger.createNewFile();
+				i.getKey().output = new BufferedOutputStream(new FileOutputStream(i.getKey().logger));
+			}
+			catch (IOException ioe) {
+				System.out.println("Could not open " + i.getKey().category + " log directory");
+				ioe.printStackTrace();
+			}
+		}
+		File globalAlias = new File(LogKitten.GLOBAL_ALIAS_PATH);
 		try {
-			if (logAlias.exists()) {
-				logAlias.delete();
+			if(globalAlias.exists()) {
+				globalAlias.delete();
 			}
-			Files.createSymbolicLink(logAlias.toPath(), file.toPath());
+			Files.createSymbolicLink(globalAlias.toPath(), global.toPath());
 		}
 		catch (IOException ioe) {
-			System.out.println("Could not alias logfile");
+			System.out.println("Could not alias global logfile");
 			ioe.printStackTrace();
 		}
+		for(Map.Entry<Kitten, String> i : kittenList.entrySet()) {
+			try {
+				if(i.getKey().loggerAlias.exists()) {
+					i.getKey().loggerAlias.delete();
+				}
+				Files.createSymbolicLink(i.getKey().loggerAlias.toPath(), i.getKey().logger.toPath());
+			}
+			catch (IOException ioe) {
+				System.out.println("Could not alias " + i.getKey().category + " logfile");
+			}
+		}
+	}//static ends
+
+	/**
+	 * Gets the mode of the robot
+	 * 
+	 * @return a string with either auton or teleop
+	 */
+	private static String getRobotMode(){
+		String robotMode = null;
+		if(DriverStation.getInstance().isAutonomous()) {
```

Comment:
Getting game-state information like this in a `static` method like this is dangerous. I don't know for certain that this won't work, so just make sure this gets tested on the bot before it gets merged.